src/js/components/DataTable/doc.js

```diff
@@ -393,6 +394,12 @@ export const themeDoc = {
     type: 'string | object',
     defaultValue: undefined,
   },
+  'dataTable.header.units': {
+    description: `The styles for the units display in the header. This
```

Comment:
```suggestion
    description: `Any Text component properties for styling the header's units text. This
```
Or something along the lines that implies the user to use Text props for `units`